instrumentation/http_instrumentation.go

```diff
@@ -102,24 +102,70 @@ func (m instrumentedHandlerFactory) NewHandler(extraLabels prometheus.Labels, ne
 }
 
 // NewInstrumentedHandlerFactory creates a new instrumentedHandlerFactory.
-func NewInstrumentedHandlerFactory(req *prometheus.Registry, hardcodedLabels []string) instrumentedHandlerFactory {
+func NewInstrumentedHandlerFactory(req *prometheus.Registry) instrumentedHandlerFactory {
```

Comment:
this helps organise the code better, but its a breaking change for anyone that relies on this package.

Reply:
I fixed all of usages 🤘

Reply:
> I fixed all the usages

This is not possible since we don't own all the code that can depend on this. We have fixed it for our usages within this project, but it is a breaking change.

Reply:
The introduction of `hardcodedLabels` here in this function was a breaking change on its own and it happened early this year. I don't think we have to worry about this also because Observatorium API didn't reach v1 yet and thus there's no guarantee of backwards compatibility in the terms of SemVer... but there isn't also a mention whether the project adheres to SemVer or not.

Reply:
👍  yeah, lets wait and see what maintainers think

Reply:
I see. Sure, maybe breaking the API won't really affect anyone practically, but this still is reducing functionality, no? So in the future, if we ever needed hardcoded labels, we need to add them to every HTTP metric again, instead of just appending to slice.

I'm not entirely sure I get what the motivation behind this refactor is and why on this middleware reorg PR. 😅 
Maybe we can iterate on this in a separate PR @douglascamata?

Reply:
The refactor is needed to fix issues with the HTTP instrumentation middleware that doesn't monitor any HTTP interaction that happens before this middleware runs. And there are plenty of other middlewares that are mounted before the HTTP instrumentation that could fail, like rate limiting.

It's a big chains of problems and solutions that cannot be solved in multiple PRs, because everything depends on each other. We would love to have a minimal PR to fix this, but even after talking a thinking a lot about this we couldn't come up with a better approach. 

I already have commits prepared to reintroduce the labeling on a per handler basis.

Reply:
We can try again to separate this, but it's hard to move a piece of the domino here without all the others falling. 😅

Reply:
Sorry if I wasn't clear. 😅 

I agree that this middleware refactor PR is necessary, but I didn't understand the reason behind removing an arg in `NewInstrumentedHandlerFactory` in this commit: https://github.com/observatorium/api/pull/430/commits/f087e9cd3b559892487476aa28c98486aae4012d?